lib/opencensus/stats/exporters/stackdriver/converter.rb

```diff
@@ -0,0 +1,269 @@
+        ##
+        # An object that converts OpenCensus stats data objects to Monitoring
+        # service protos
+        #
+        # @private
+        #
+        class Converter
+          ##
+          # Create a converter
+          #
+          # @param [String] project_id Google project ID
+          #
+          def initialize project_id
+            @project_id = project_id
+          end
+
+          # Convert view to metric descriptor
+          #
+          # @param [OpenCensus::Stats:View] view Stats view
+          # @param [String] metric_prefix Metric prefix name
+          # @return [Google::Api::MetricDescriptor]
+          #
+          def convert_metric_descriptor view, metric_prefix
+            descriptor = Google::Api::MetricDescriptor.new(
+              type: make_metric_type(metric_prefix, view.name),
+              display_name: view.measure.name,
+              metric_kind: convert_metric_kind(view.aggregation),
+              value_type: convert_metric_value_type(view),
+              unit: view.measure.unit,
+              labels: convert_labels(view.columns)
+            )
+
+            descriptor.description ||= view.description
+            descriptor
+          end
+
+          # Conver to lables
+          #
+          # @param [Array<String>] names
+          # @return [Array<Google::Api::LabelDescriptor>]
+          def convert_labels names
+            names.map do |name|
+              Google::Api::LabelDescriptor.new(
+                key: name,
+                value_type: Google::Api::LabelDescriptor::ValueType::STRING
+              )
+            end
+          end
+
+          # Convert to metric view type.
+          #
+          # @param [OpenCensus::Stats:View] view Stats view
+          # @param [Symbol]
+          #
+          def convert_metric_value_type view
+            case view.aggregation
+            when OpenCensus::Stats::Aggregation::Distribution
+              Google::Api::MetricDescriptor::ValueType::DISTRIBUTION
+            when OpenCensus::Stats::Aggregation::Count
+              Google::Api::MetricDescriptor::ValueType::INT64
+            when OpenCensus::Stats::Aggregation::Sum,
+                OpenCensus::Stats::Aggregation::LastValue
+              if view.measure.int64?
+                Google::Api::MetricDescriptor::ValueType::INT64
+              else
+                Google::Api::MetricDescriptor::ValueType::DOUBLE
+              end
+            end
+          end
+
+          # Convert to metric kind
+          #
+          # @param [OpenCensus::Stats:Aggregation::LastValue,
+          #   OpenCensus::Stats:Aggregation::Sum,
+          #   OpenCensus::Stats:Aggregation::Count,
+          #   OpenCensus::Stats:Aggregation::Distribution] aggregation
+          #   Aggregation type
+          # @param [Symbol]
+          #
+          def convert_metric_kind aggregation
+            if aggregation.instance_of? OpenCensus::Stats::Aggregation:: \
+              LastValue
```

Comment:
This line breaking is very awkward. If you're having trouble with line length, try setting a local variable (e.g. `last_value_class = OpenCensus::Stats::Aggregation::LastValue`)